src/app/activity/bounty/Claim.ts

```diff
@@ -68,56 +100,147 @@ const getDbHandler = async (request: ClaimRequest): Promise<{dbBountyResult: Bou
+                    Log.error('failed to update evergreen bounty with deleted status');
+                    throw new Error('Sorry something is not working, our devs are looking into it.');
+                }
+            }
+        }
+    } else {
+        claimedBounty = dbBountyResult;
+    }
+ 
+    const writeResult: UpdateWriteOpResult = await bountyCollection.updateOne(claimedBounty, {
+        $set: {
+            claimedBy: {
+                discordHandle: claimedByUser.user.tag,
+                discordId: claimedByUser.user.id,
+                iconUrl: claimedByUser.user.avatarURL(),
+            },
+            claimedAt: currentDate,
+            status: BountyStatus.in_progress,
+        },
+        $push: {
+            statusHistory: {
+                status: BountyStatus.in_progress,
+                setAt: currentDate,
+            },
+        },
+    });
 
     if (writeResult.result.ok !== 1) {
-        Log.error(`Write result did not execute correctly`);
+        Log.error('failed to update claimed bounty with in progress status');
         throw new Error(`Write to database for bounty ${request.bountyId} failed for ${request.activity} `);
     }
+
+    return claimedBounty;
 }
 
-export const claimBountyMessage = async (message: Message, claimedByUser: GuildMember): Promise<any> => {
-	Log.debug(`fetching bounty message for claim`)
+export const claimBountyMessage = async (message: Message, claimedBounty: BountyCollection, claimedByUser: GuildMember, originalBounty: BountyCollection): Promise<any> => {
+    Log.debug(`fetching bounty message for claim`)
     
-    const embedMessage: MessageEmbed = message.embeds[0];
+    const existingEmbeds = message.embeds[0];
+    const embedNewMessage: MessageEmbed = new MessageEmbed(existingEmbeds);
 
-	embedMessage.fields[BountyEmbedFields.status].value = BountyStatus.in_progress;
-	embedMessage.setColor('#d39e00');
-	embedMessage.addField('Claimed by', claimedByUser.user.tag, true);
+    // Send claimed bounty by DM
+    embedNewMessage.fields[BountyEmbedFields.status].value = BountyStatus.in_progress;
+    embedNewMessage.fields[BountyEmbedFields.bountyId].value = claimedBounty._id.toString();
+    embedNewMessage.setTitle(BountyUtils.createPublicTitle(<Bounty>claimedBounty));
+    embedNewMessage.setURL(process.env.BOUNTY_BOARD_URL + claimedBounty._id.toString());
+    embedNewMessage.setColor('#d39e00');
+    embedNewMessage.addField('Claimed by', claimedByUser.user.tag, true);
+    embedNewMessage.setFooter({text: '📮 - submit | 🆘 - help'});
+    const claimantMessage: Message = await claimedByUser.send({ embeds: [embedNewMessage] });
+    await addClaimReactions(claimantMessage);
 
-	embedMessage.setFooter('📮 - submit | 🆘 - help');
-	await message.edit({ embeds: [embedMessage] });
-	await addClaimReactions(message);
+    // If Bounty status is no longer open, delete the board message, otherwise update title
+    if (originalBounty.status !== BountyStatus.open) {
```

Comment:
something about this makes me uneasy. But I can't explain it, so it could just be BS. can we check more specifically for `status !== BountyStatus.deleted`? What do you think?

Reply:
With this new flow, there should be no non-open bounties in the board channel. There may be a case where there was a non-open bounty left in the channel from the old bot, and this should clean that up, which we can do because new messages will be in the DMs. We need to verify this with both bots installed.